packages/flutter_tools/lib/src/bundle.dart

```diff
@@ -72,73 +66,30 @@ class BundleBuilder {
     List<String> fileSystemRoots,
     String fileSystemScheme,
-    bool shouldBuildWithAssemble = false,
   }) async {
     mainPath ??= defaultMainPath;
     depfilePath ??= defaultDepfilePath;
     assetDirPath ??= getAssetBuildDirectory();
     packagesPath ??= fs.path.absolute(PackageMap.globalPackagesPath);
-    applicationKernelFilePath ??= getDefaultApplicationKernelPath(trackWidgetCreation: trackWidgetCreation);
     final FlutterProject flutterProject = FlutterProject.current();
 
-    if (shouldBuildWithAssemble) {
-      await buildWithAssemble(
-        buildMode: buildMode ?? BuildMode.debug,
-        targetPlatform: platform,
-        mainPath: mainPath,
-        flutterProject: flutterProject,
-        outputDir: assetDirPath,
-        depfilePath: depfilePath,
-        precompiled: precompiledSnapshot,
-      );
-      return;
-    }
-
-    DevFSContent kernelContent;
-    if (!precompiledSnapshot) {
-      if ((extraFrontEndOptions != null) && extraFrontEndOptions.isNotEmpty) {
-        printTrace('Extra front-end options: $extraFrontEndOptions');
-      }
-      ensureDirectoryExists(applicationKernelFilePath);
-      final KernelCompiler kernelCompiler = await kernelCompilerFactory.create(flutterProject);
-      final CompilerOutput compilerOutput = await kernelCompiler.compile(
-        sdkRoot: artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath, mode: buildMode),
-        mainPath: fs.file(mainPath).absolute.path,
-        outputFilePath: applicationKernelFilePath,
-        depFilePath: depfilePath,
-        enableAsserts: buildMode == BuildMode.debug,
-        trackWidgetCreation: trackWidgetCreation,
-        extraFrontEndOptions: extraFrontEndOptions,
-        fileSystemRoots: fileSystemRoots,
-        fileSystemScheme: fileSystemScheme,
-        packagesPath: packagesPath,
-      );
-      if (compilerOutput?.outputFilename == null) {
-        throwToolExit('Compiler failed on $mainPath');
-      }
-      kernelContent = DevFSFileContent(fs.file(compilerOutput.outputFilename));
-
-      fs.directory(getBuildDirectory()).childFile('frontend_server.d')
-          .writeAsStringSync('frontend_server.d: ${artifacts.getArtifactPath(Artifact.frontendServerSnapshotForEngineDartSdk)}\n');
-    }
-
-    final AssetBundle assets = await buildAssets(
-      manifestPath: manifestPath,
-      assetDirPath: assetDirPath,
-      packagesPath: packagesPath,
-      reportLicensedPackages: reportLicensedPackages,
+    await buildWithAssemble(
+      buildMode: buildMode ?? BuildMode.debug,
+      targetPlatform: platform,
+      mainPath: mainPath,
+      flutterProject: flutterProject,
+      outputDir: assetDirPath,
+      depfilePath: depfilePath,
+      precompiled: precompiledSnapshot,
     );
-    if (assets == null) {
-      throwToolExit('Error building assets', exitCode: 1);
+    // Work around for flutter_tester placing kernel artifacts in odd places.
+    if (applicationKernelFilePath != null) {
```

Comment:
flutter_tester is currently the only use of this logic. In the long term, we should clean this up to use a tempDir or something similar.

Reply:
Is there an issue open for this?